route/build/src/remap.f90

```diff
@@ -260,49 +260,69 @@ subroutine remap_1D_runoff(runoff_data_in, remap_data_in, basinRunoff, ierr, mes
+                         basin_flux_sort,  & ! output: sorted input states and fluxes based on ID_total_in (the same size as ID_total_in)
+                         ierr, message)
   implicit none
   ! input
-  type(runoff)         , intent(in)  :: runoff_data_in   ! runoff for one time step for all HRUs
+  integer(i4b)         , intent(in)    :: ID_in(:)            ! input: the array of ids for HRU/seg (can be from 1 to nHRU or nSeg)
+  integer(i4b)         , intent(in)    :: IX_in(:)            ! input: the array of location of ids for HRU/seg in ID_total_in (can be from 1 to nHRU or nSeg)
+  real(dp)             , intent(in)    :: flux_in(:)          ! input: the array of input fluxes, should be the same size as ID_in
+  integer(i4b)         , intent(in)    :: ID_total_in(:)      ! input: the sorted array of ids for HRU/seg (should be with nHRU or nSeg)
+  ! input/output
+  real(dp)             , intent(inout) :: basin_flux_sort(:)  ! inout: sorted input states and fluxes based on ID_total_in (the same size as ID_total_in)
   ! output
-  real(dp)             , intent(out) :: basinRunoff(:)   ! basin runoff
-  integer(i4b)         , intent(out) :: ierr             ! error code
-  character(len=strLen), intent(out) :: message          ! error message
+  integer(i4b)         , intent(out)   :: ierr             ! error code
+  character(len=strLen), intent(out)   :: message          ! error message
   ! local
-  integer(i4b)                       :: iHRU,jHRU        ! index of basin in the routing layer
-  real(dp)    , parameter            :: xTol=1.e-6_dp    ! tolerance to avoid divide by zero
-  integer(i4b), parameter            :: ixCheck=-huge(iHRU) ! basin to check
+  integer(i4b)                         :: i,j              ! index of basin in the routing layer
 
-  ierr=0; message="sort_runoff/"
+  ierr=0; message="sort_flux/"
 
-  ! loop through hrus in the runoff layer
-  do iHRU=1,size(runoff_data_in%hru_ix)
+!  ! allocate basin_flux_sort is not allocated
+!  if (.not.allocated(basin_flux_sort)) then
+!    ierr = 10
+!    if(ierr/=0)then; message=trim(message)//'basin_flux_sort is not allocated'; return; endif
+!  endif
 
-   ! define the HRU index in the routing vector
-   jHRU = runoff_data_in%hru_ix(iHRU)
+!  ! the size of basin_flux_sort and ID_total_in
+!  if (size(basin_flux_sort)/=size(ID_total_in)) then
+!    ierr = 20
+!    if(ierr/=0)then; message=trim(message)//'basin_flux_sort and ID are not the same size'; return; endif
+!  endif
+
+!  ! check the size of ID_in, IX_in and flux_in
+!  if ((size(ID_in)/=size(IX_in)).or.(size(IX_in)/=size(flux_in)).or.(size(flux_in)/=size(ID_in))) then
+!    ierr = 30
+!    if(ierr/=0)then; message=trim(message)//'ID_in, IX_in and flux_in not the same size'; return; endif
+!  endif
+
+  ! initializing the basin_flux_sort to zero, assuming non existing elements are all set to zero
+  basin_flux_sort = 0._dp
```

Comment:
Thinking of sorting for only selected segID/hruID, I think this is better to be initialized as realMissing since 0._dp might be valid value.  Once it is sorted into 0 initialed array, it might cause silent, runtime logical error.